From 5dbaf8eebc5b66230e0131b09651c7e40bf0e9de Mon Sep 17 00:00:00 2001
From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Date: Tue, 20 Aug 2019 21:41:16 +0200
Subject: [PATCH] acinclude/os-deps.m4: fix cross-compilation

Do not check check file descriptor maximum value through AC_RUN_IFELSE
when cross-compiling as this will raise an error

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
[Upstream status: https://github.com/squid-cache/squid/pull/464]
---
 acinclude/os-deps.m4 | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/acinclude/os-deps.m4 b/acinclude/os-deps.m4
index b50717517..ec10a54c6 100644
--- a/acinclude/os-deps.m4
+++ b/acinclude/os-deps.m4
@@ -169,7 +169,9 @@ AC_MSG_CHECKING(Maximum number of filedescriptors we can open)
 SQUID_STATE_SAVE(maxfd)
 dnl FreeBSD pthreads break dup2().
   AS_CASE([$host_os],[freebsd],[ LDFLAGS=`echo $LDFLAGS | sed -e "s/-pthread//"` ])
-  AC_RUN_IFELSE([AC_LANG_SOURCE([[
+  dnl AC_RUN_IFELSE can't be run when cross-compiling
+  AS_CASE([$cross_compiling],[no],[
+    AC_RUN_IFELSE([AC_LANG_SOURCE([[
 #include <stdio.h>
 #include <unistd.h>
 #include <stdlib.h>
@@ -231,7 +233,8 @@ int main(int argc, char **argv) {
 	fprintf (fp, "%d\n", i & ~0x3F);
 	return 0;
 }
-  ]])],[squid_filedescriptors_limit=`cat conftestval`],[],[])
+    ]])],[squid_filedescriptors_limit=`cat conftestval`],[],[])
+  ])
   dnl Microsoft MSVCRT.DLL supports 2048 maximum FDs
   AS_CASE(["$host_os"],[mingw|mingw32],[squid_filedescriptors_limit="2048"])
   AC_MSG_RESULT($squid_filedescriptors_limit)
-- 
2.20.1

